sul-dlss/sdr-preservation-core
# [WIP?] rebase develop on master so it can be merged cleanly to master

here's an attempt at rebasing develop on master.  took a little while, but it was pretty mechanical, and didn't really require any hard decisions, so hopefully it works?  this was my procedure:
* pull the latest sdr-preservation-core code
* checkout `develop`
* checkout a new branch off of develop (`develop_try-rebase-2017-06-26`)
* `git rebase master`, and deal with conflicts 1 by 1
  * luckily, with the exception of Gemfile.lock, which i didn't look at on either branch, every "conflicting" file i ran into actually had the same contents on current `master` and current `develop`, with the exception in one or two cases of a trailing newline.  to deal with Gemfile.lock conflicts, i first resolved the Gemfile conflict, then just deleted Gemfile.lock, and did a `bundle install` to regenerate it.  as the process went on, more and more of the "conflicts" resulted in the `No changes - did you forget to use 'git add'? If there is nothing left to stage, chances are that something else already introduced the same changes; you might want to skip this patch.` at the end of each resolution step.  in which case i skipped and moved on.

anyway, this branch and its PR are the result of that process.  hopefully that's helpful in getting develop merged.

you'll notice that some of the commits in this PR don't really do anything.  i assume that's a result of the above rebasing process.  i'm happy to pull them out of the PR or clean them up, but i wanted to start out with this branch being the result of the above totally mechanical process.
